File: app/src/main/java/com/youtility/customview/CustomButton.java
package com.youtility.customview;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.Button;


/**
 * Created by Mohit Rajput on 11/12/17.
 * Customized and common button
 */
@SuppressLint("AppCompatCustomView")
public class CustomButton extends Button {

    public CustomButton(Context context) {
        super(context);
        init(null, 0);
    }

    public CustomButton(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(attrs, 0);
    }

    public CustomButton(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init(attrs, defStyle);
    }

    @SuppressWarnings("unused")
    private void init(AttributeSet attrs, int defStyle) {
        // Load attributes
        setAllCaps(false);
        // Update TextPaint and text measurements from attributes
        invalidateTextPaintAndMeasurements();
    }

    private void invalidateTextPaintAndMeasurements() {

    }

    @Override
    public void setEnabled(boolean enabled) {
        if (enabled) {
            setAlpha(1.0f);
        } else {
            setAlpha(0.6f);
        }
        super.setEnabled(enabled);
    }
}

File: energyswitchcui/src/main/kotlin/com/youtility/energyswitchcui/customview/CustomButton.kt
package com.youtility.energyswitchcui.customview

import android.content.Context
import android.support.v7.widget.AppCompatButton
import android.util.AttributeSet


/**
 * Created by Mohit Rajput on 3/4/18.
 * Customized and common button
 */
class CustomButton : AppCompatButton {

    constructor(context: Context) : super(context) {
        init(null, 0)
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init(attrs, 0)
    }

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {
        init(attrs, defStyle)
    }

    @Suppress("UNUSED_PARAMETER")
    private fun init(attrs: AttributeSet?, defStyle: Int) {
        // Load attributes
        isAllCaps = false
        // Update TextPaint and text measurements from attributes
        invalidateTextPaintAndMeasurements()
    }

    private fun invalidateTextPaintAndMeasurements() {

    }

    override fun setEnabled(enabled: Boolean) {
        if (enabled) {
            alpha = 1.0f
        } else {
            alpha = 0.6f
        }
        super.setEnabled(enabled)
    }
}
